i18n(it): tutorial setup - page 4 and 5 (#4944)
withastro/docs: src/content/docs/it/tutorial/1-setup/4.mdx

---
type: tutorial
title: Salva il tuo codice in un repository online
description: |-
  Tutorial: Costruisci il tuo primo blog Astro —
  Crea un repository GitHub per il progetto del tutorial

> i18nReady: true

Reviewer: ```suggestion

```

This is just for the primary version... it's not really clear from the variable itself.

<PreCheck>
  - Salva online il repository del tuo progetto
</PreCheck>

Questo tutorial utilizza GitHub per salvare il repository e collegarlo a un servizio di hosting. È comunque possibile utilizzare altri servizi git online, se lo desideri.

:::note
Se già usi git e sai come creare un repository su GitHub, puoi creare il tuo repository ora utilizzando la modalità con cui sei più a tuo agio. Salta pure alla pagina successiva ["Pubblica il tuo sito sul web"](/it/tutorial/1-setup/5/). 
:::

## Creare un repository su GitHub

Esistono modi diversi per salvare su GitHub il codice che hai in locale, tuttavia in questo tutorial useremo un metodo che non richiede di usare git dalla linea di comando. 

1. Accedi con le tue credenziali a GitHub.com con il browser e fai click sul <kbd>+</kbd> nell'angolo in alto a destra dello schermo per creare un nuovo repository. 

2. Scegli un nome per il tuo repository. Non deve necessariamente coincidere col nome della cartella del progetto.

3. Ti verranno mostrate alcune opzioni, ma non è necessario modificare i valori di default. Scorri in basso e fai clic sul pulsante <kbd>Create Repository</kbd>.

> 4. Ti verranno presentate altre opzioni per i passi successivi, ma non ne avrai bisogno, semplicemente prendi nota dell'URL del tuo repository. A questo  punto puoi uscire dalla pagina senza fare nient'altro.

Reviewer 2: Whitespace removal
```suggestion
4. Ti verranno presentate altre opzioni per i passi successivi, ma non ne avrai bisogno, semplicemente prendi nota dell'URL del tuo repository. A questo punto puoi uscire dalla pagina senza fare nient'altro.
```

## Salvare (commit) il codice locale su GitHub

> Nella sezione precedente hai fatto alcune modifiche al contenuto della pagina. Questo significa che i file del tuo progetto sono cambiati, quindi VS Code mosterà un numero sopa l'icona "Source" nel menu. La scheda "Source" ti sarà utile per fare il commit dei tuoi file periodicamente su GitHub.

Reviewer: ```suggestion
Nella sezione precedente hai fatto alcune modifiche al contenuto della pagina. Questo significa che i file del tuo progetto sono cambiati, quindi VS Code mosterà un numero sopa l'icona "Sorgente" nel menu. La scheda "Sorgente" ti sarà utile per fare il commit dei tuoi file periodicamente su GitHub.
```

VS Code is available in Italian, so we translate the menu names.

> 1. Fai clic sulla scheda "Source Control" in VS Code per vedere un elenco di file che sono cambiati. Se vedi un messaggio che ti chiede di installare `git`, segui le istruzioni e poi riavvia VS Code.

Reviewer: ```suggestion
1. Fai clic sulla scheda "Controllo del codice sorgente" in VS Code per vedere un elenco di file che sono cambiati. Se vedi un messaggio che ti chiede di installare `git`, segui le istruzioni e poi riavvia VS Code.
```

See above.

> 2. Fai clic sul menu <kbd>•••</kbd> "3 puntini" appena sopra il messaggio di commit e scegli <kbd>Remote</kbd> > <kbd>Add Remote</kbd>.

Reviewer: ```suggestion
2. Fai clic sul menu <kbd>•••</kbd> "3 puntini" appena sopra il messaggio di commit e scegli <kbd>Repository remoto</kbd> > <kbd>Aggiungi repository remoto</kbd>.
```

> 3. Selezione <kbd>Add remote from GitHub</kbd>. Se necessario autenticati seguento le istruzioni, poi ritorna a VS Code e ripeti questa azione.

Reviewer 2: There are some typos:

```suggestion
3. Seleziona <kbd>Add remote from GitHub</kbd>. Se necessario autenticati seguendo le istruzioni, poi ritorna a VS Code e ripeti questa azione.
```

> 3. Selezione <kbd>Add remote from GitHub</kbd>. Se necessario autenticati seguento le istruzioni, poi ritorna a VS Code e ripeti questa azione.

Reviewer: ```suggestion
3. Seleziona <kbd>Aggiungi repository remoto da GitHub</kbd>. Se necessario autenticati seguendo le istruzioni, poi ritorna a VS Code e ripeti questa azione.
```

4. Dovresti vedere un elenco di tutti i tuoi repository su GitHub. Seleziona quello che hai creato per questo progetto. Se non vedi il tuo progetto, incolla direttamente il suo URL di GitHub. Potrebbe venirti chiesto di dare un nome al repository locale. Puoi scegliere il nome che preferisci. 

5. Nella parte superiore del pannello è possibile inserire un **messaggio di commit** (cioè la descrizione dei cambiamenti ai tuoi file). Scrivi `commit iniziale` e fai clic sul pulsante <kbd>Commit</kbd> per fare il commit delle tue modifiche.

> 6. Potresti vedere un messaggio che ti informa che non hai "staged commits", e che ti chiede se vuoi inserirli nell'area di stage. Fai clic su <kbd>Always</kbd> e continua.

Reviewer: ```suggestion
6. Potresti vedere un messaggio che ti informa che non hai "commit preparate", e che ti chiede se vuoi inserirli nell'area di preparazione. Fai clic su <kbd>Sempre</kbd> e continua.
```

> 7. Infine, l'elenco dei file modificato dovrebbe essere sostituito da un pulsante <kbd>Publish</kbd>. Fai clic per inviare i tuoi file salavati su GitHub.

Reviewer: ```suggestion
7. Infine, l'elenco dei file modificato dovrebbe essere sostituito da un pulsante <kbd>Pubblica</kbd>. Fai clic per inviare i tuoi file salvati su GitHub.
```

### Visualizza il tuo progetto su GitHub

> Per verificare se il tuo progetto è stato salvato su GitHub, visita GitHub.com e cerca all'interno del tuo account l'elenco dei tuoi repository. Scegli l'ultimo che hai creato e verifica che contiene i file del tuo progetto Astro.

Reviewer 2: ```suggestion
Per verificare se il tuo progetto è stato salvato su GitHub, visita GitHub.com e cerca all'interno del tuo account l'elenco dei tuoi repository. Scegli l'ultimo che hai creato e verifica che contenga i file del tuo progetto Astro.
```

<Box icon="check-list">
## Checklist

<Checklist>
- [ ] Ho salvato il mio progetto su GitHub.
</Checklist>
</Box>

### Resources

> - <p>[Utilizzare il sistema di controllo delle versioni Git in VS Code](https://code.visualstudio.com/docs/sourcecontrol/overview#_git-support) <Badge>external</Badge></p>

Reviewer 2: ```suggestion
- <p>[Utilizzare il sistema di versionamento Git in VS Code](https://code.visualstudio.com/docs/sourcecontrol/overview#_git-support) <Badge>external</Badge></p>
```

> - <p>[Utilizzare il sistema di controllo delle versioni Git in VS Code](https://code.visualstudio.com/docs/sourcecontrol/overview#_git-support) <Badge>external</Badge></p>

Reviewer: ```suggestion
- <p>[Utilizzare il sistema di versionamento Git in VS Code](https://code.visualstudio.com/docs/sourcecontrol/overview#_git-support) <Badge>esterno</Badge></p>
```
